larg4/pluginActions/MCTruthEventAction_service.cc

```diff
@@ -129,27 +111,27 @@ void larg4::MCTruthEventActionService::generatePrimaries(G4Event * anEvent) {
         // Get the particle table if necessary.  (Note: we're
         // doing this "late" because I'm not sure at what point
         // the G4 particle table is initialized in the loading process.
-        if ( fParticleTable == 0 ){
+        if (fParticleTable == nullptr) {
           fParticleTable = G4ParticleTable::GetParticleTable();
         }
 
         // Get Geant4's definition of the particle.
-        G4ParticleDefinition* particleDefinition;
+        G4ParticleDefinition* particleDefinition = nullptr;
 
-        if(pdgCode==0){
+        if (pdgCode == 0) {
           particleDefinition = fParticleTable->FindParticle("opticalphoton");
         }
         else
           particleDefinition = fParticleTable->FindParticle(pdgCode);
 
         if ( pdgCode > 1000000000) { // If the particle is a nucleus
```

Comment:
this really should be 
 if ( pdgCode > 1000000000 &&  pdgCode < 2000000000) 
 and a separate handling of PDG >=  2000000000 dealt with (i.e. notification that they can't be handled as they're generator specific and not ions)

Reply:
"this" being line 145/127

Reply:
@nusense, thanks for point that out.  @lgarren, although Robert's comment is not directly related to the changes I've made here, do you think there would be any problem with me adjusting the code along the lines he suggests?  I'm guessing the code wouldn't be working correctly anyway if a pdgCode of 2000000000 or greater were encountered...?

Reply:
As long as the change doesn't break anything, I see no problem.